File: responser-backend/src/main/java/com/responser/backend/controller/payload/PageableResponse.java
package com.responser.backend.controller.payload;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class PageableResponse<T> {

    private Long totalElements;

    private Integer totalPages;

    private Integer currentPageNumber;

    private Integer currentPageSize;

    private Boolean isLast;

    private Boolean isFirst;

    private Boolean hasPrevious;

    private Boolean hasNext;

    private List<T> data;
}

File: reviewly-console/src/app/model/PageableResponse.ts
export class PageableResponse<T> {

    totalElements: number;

    totalPages: number;

    currentPageNumber: number;

    currentPageSize: number;

    isLast: boolean;

    data: T[];

    public getFirst = (): T => {
        return this.data[0]
    }

    public static deserialize<T>(data: any, dataList: T[]): PageableResponse<T> {
        return Object.assign(new PageableResponse<T>(), data, {
            data: dataList
        });
    }
}
